### PR TITLE
Banner place picker

### PR DESCRIPTION
**Pull request by matejminar**
Hey guys, 

I’ve got the target element selector for banner placement functionality ready.

Few concerns:
1. as we discussed, I placed the CLIENT_SITE_URL (the site which opens, when user clicks on the target element selector button) in the .env file - but what about the scenario, when user wants to place banner somewhere else than homepage?

2. in the remp/Campaign/resources/assets/js/bannerSelector.js line 26 I am setting the allowed origin from which the client site (where the selector shows) is allowed to listen. We should probably take this value from somewhere else, or will editing this line be part of required initial configuration?

3. in Campaign/resources/assets/js/remplib.js line 98 I load the bannerSelector script. I would like to pass the second argument (callback) to init the function(because now it inits itself upon loading), maybe then we could pass the allowedOrigin parameter from the above concern - how do I pass the init callback(RempBannerSelector.init) to the second parameter without getting an error, because it’s not loaded yet?

### CODE REVIEW

File: Campaign/resources/assets/js/bannerSelector.js [hunk 1]
```diff
@@ -0,0 +1,848 @@
...
+      element.className.split(' ').indexOf(classname) >= 0
+    )
+      return true;
+    return (
+      element.parentNode && hasSomeParentTheClass(element.parentNode, classname)
+    );
+  }
+  function injectBanner() {
+    hideBannerInsertionSettingsModal();
+    alert('TODO, insert markup');
```
Comment (rootpd) [inline Campaign/resources/assets/js/bannerSelector.js]: I think we just remove this for now as it's mostly commented code and a TODO.

File: Campaign/resources/assets/js/bannerSelector.js [hunk 2]
```diff
@@ -0,0 +1,848 @@
...
+    cancelAndReturnToRempElement,
+    saveAndReturnToRempElement;
+
+  // window reference from which this site was open
+  var rempParentWebsite;
+  // TODO: move this to some sort of config file, or accept it during init?
+  var allowedOrigin = 'http://campaign.remp.press';
+
+  // markup for injected toolbar
+  var toolbarMarkup = `
```
Comment (rootpd) [inline Campaign/resources/assets/js/bannerSelector.js]: Btw. seeing this as a markup and style in JS looks very like this should have been a Vue component. Nothing to address, just for a future reference.

**Comment by rootpd** [comment]
One more thing I realized during testing. In the original task there was a nice to have feature, to highlight elements with specific class and set them minimal height if their height is zero so they're visible and clickable (most common use-case for banner positions).

I'd like to come back to this and update it slightly. If you can, please add the feature in a way, that it will highlight (and set minimal height) to all elements matching provided selector on the target website.

The selector should come from the `remplib.campaign[.bannerSelector.highlight]` config object and if it's not present, it should default to `.remp-banner`.

**Comment by matejminar** [comment]
Thanks a lot for feedback and guidance!
I've updated the PR.

File: Campaign/resources/views/banners/_form.blade.php [hunk 3]
```diff
@@ -27,6 +27,8 @@
 
     remplib.bannerForm.bind("#banner-form", banner);
 
+    var clientSiteUrl = '{{ Config::get('app.client_site_url') }}';
```
Comment (markoph) [inline Campaign/resources/views/banners/_form.blade.php]: If `clientSiteUrl` is not set, user is redirected to same URL with fragment identifier bannerPicker. No message with info or error what happend. Maybe we could hide (or disable) button if `clientSiteUrl` is empty?

File: Campaign/package.json [hunk 4]
```diff
@@ -35,6 +35,7 @@
     "material-design-iconic-font": "^2.2.0",
     "moment": "^2.20.1",
     "node-waves": "^0.7.5",
+    "optimal-select": "^4.0.1",
```
Comment (markoph) [inline Campaign/package.json]: Please push also `yarn.lock` file.

**Comment by matejminar** [comment]
Roger that. PR is updated

**Pull request merged**